r2s编译出错

详细日志

A：make[6]: Entering directory '/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9'
[  0%] Building C object library/CMakeFiles/mbedcrypto_static.dir/armv8ce_aes.c.o
In file included from /home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:30:
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c: In function 'mbedtls_armv8ce_aes_crypt_ecb':
/home/ling/lede/staging_dir/toolchain-aarch64_generic_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/include/arm_neon.h:12440:1: error: inlining failed in call to always_inline 'vaesmcq_u8': target specific option mismatch
 vaesmcq_u8 (uint8x16_t data)
 ^~~~~~~~~~
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:62:17: note: called from here
             x = vaesmcq_u8( x );
                 ^~~~~~~~~~~~~~~
In file included from /home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:30:
/home/ling/lede/staging_dir/toolchain-aarch64_generic_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/include/arm_neon.h:12426:1: error: inlining failed in call to always_inline 'vaeseq_u8': target specific option mismatch
 vaeseq_u8 (uint8x16_t data, uint8x16_t key)
 ^~~~~~~~~
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:61:17: note: called from here
             x = vaeseq_u8( x, k );
                 ^~~~~~~~~~~~~~~~~
In file included from /home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:30:
/home/ling/lede/staging_dir/toolchain-aarch64_generic_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/include/arm_neon.h:12440:1: error: inlining failed in call to always_inline 'vaesmcq_u8': target specific option mismatch
 vaesmcq_u8 (uint8x16_t data)
 ^~~~~~~~~~
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:62:17: note: called from here
             x = vaesmcq_u8( x );
                 ^~~~~~~~~~~~~~~
In file included from /home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:30:
/home/ling/lede/staging_dir/toolchain-aarch64_generic_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/include/arm_neon.h:12426:1: error: inlining failed in call to always_inline 'vaeseq_u8': target specific option mismatch
 vaeseq_u8 (uint8x16_t data, uint8x16_t key)
 ^~~~~~~~~
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:61:17: note: called from here
             x = vaeseq_u8( x, k );
                 ^~~~~~~~~~~~~~~~~
In file included from /home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:30:
/home/ling/lede/staging_dir/toolchain-aarch64_generic_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/include/arm_neon.h:12426:1: error: inlining failed in call to always_inline 'vaeseq_u8': target specific option mismatch
 vaeseq_u8 (uint8x16_t data, uint8x16_t key)
 ^~~~~~~~~
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:66:13: note: called from here
         x = vaeseq_u8( x, k );
             ^~~~~~~~~~~~~~~~~
In file included from /home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:30:
/home/ling/lede/staging_dir/toolchain-aarch64_generic_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/include/arm_neon.h:12426:1: error: inlining failed in call to always_inline 'vaeseq_u8': target specific option mismatch
 vaeseq_u8 (uint8x16_t data, uint8x16_t key)
 ^~~~~~~~~
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:66:13: note: called from here
         x = vaeseq_u8( x, k );
             ^~~~~~~~~~~~~~~~~
In file included from /home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:30:
/home/ling/lede/staging_dir/toolchain-aarch64_generic_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/include/arm_neon.h:12447:1: error: inlining failed in call to always_inline 'vaesimcq_u8': target specific option mismatch
 vaesimcq_u8 (uint8x16_t data)
 ^~~~~~~~~~~
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:75:17: note: called from here
             x = vaesimcq_u8( x );
                 ^~~~~~~~~~~~~~~~
In file included from /home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:30:
/home/ling/lede/staging_dir/toolchain-aarch64_generic_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/include/arm_neon.h:12433:1: error: inlining failed in call to always_inline 'vaesdq_u8': target specific option mismatch
 vaesdq_u8 (uint8x16_t data, uint8x16_t key)
 ^~~~~~~~~
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:74:17: note: called from here
             x = vaesdq_u8( x, k );
                 ^~~~~~~~~~~~~~~~~
In file included from /home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:30:
/home/ling/lede/staging_dir/toolchain-aarch64_generic_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/include/arm_neon.h:12447:1: error: inlining failed in call to always_inline 'vaesimcq_u8': target specific option mismatch
 vaesimcq_u8 (uint8x16_t data)
 ^~~~~~~~~~~
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:75:17: note: called from here
             x = vaesimcq_u8( x );
                 ^~~~~~~~~~~~~~~~
In file included from /home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:30:
/home/ling/lede/staging_dir/toolchain-aarch64_generic_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/include/arm_neon.h:12433:1: error: inlining failed in call to always_inline 'vaesdq_u8': target specific option mismatch
 vaesdq_u8 (uint8x16_t data, uint8x16_t key)
 ^~~~~~~~~
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:74:17: note: called from here
             x = vaesdq_u8( x, k );
                 ^~~~~~~~~~~~~~~~~
In file included from /home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:30:
/home/ling/lede/staging_dir/toolchain-aarch64_generic_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/include/arm_neon.h:12433:1: error: inlining failed in call to always_inline 'vaesdq_u8': target specific option mismatch
 vaesdq_u8 (uint8x16_t data, uint8x16_t key)
 ^~~~~~~~~
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:79:13: note: called from here
         x = vaesdq_u8( x, k );
             ^~~~~~~~~~~~~~~~~
In file included from /home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:30:
/home/ling/lede/staging_dir/toolchain-aarch64_generic_gcc-8.4.0_musl/lib/gcc/aarch64-openwrt-linux-musl/8.4.0/include/arm_neon.h:12433:1: error: inlining failed in call to always_inline 'vaesdq_u8': target specific option mismatch
 vaesdq_u8 (uint8x16_t data, uint8x16_t key)
 ^~~~~~~~~
/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/library/armv8ce_aes.c:79:13: note: called from here
         x = vaesdq_u8( x, k );
             ^~~~~~~~~~~~~~~~~
library/CMakeFiles/mbedcrypto_static.dir/build.make:131: recipe for target 'library/CMakeFiles/mbedcrypto_static.dir/armv8ce_aes.c.o' failed
make[6]: *** [library/CMakeFiles/mbedcrypto_static.dir/armv8ce_aes.c.o] Error 1
make[6]: Leaving directory '/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9'
CMakeFiles/Makefile2:497: recipe for target 'library/CMakeFiles/mbedcrypto_static.dir/all' failed
make[5]: *** [library/CMakeFiles/mbedcrypto_static.dir/all] Error 2
make[5]: Leaving directory '/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9'
Makefile:146: recipe for target 'all' failed
make[4]: *** [all] Error 2
make[4]: Leaving directory '/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9'
Makefile:173: recipe for target '/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/.built' failed
make[3]: *** [/home/ling/lede/build_dir/target-aarch64_generic_musl/mbedtls-2.16.9/.built] Error 2
make[3]: Leaving directory '/home/ling/lede/package/libs/mbedtls'
time: package/libs/mbedtls/compile#0.33#0.06#0.37
package/Makefile:111: recipe for target 'package/libs/mbedtls/compile' failed
make[2]: *** [package/libs/mbedtls/compile] Error 2
make[2]: Leaving directory '/home/ling/lede'
package/Makefile:107: recipe for target '/home/ling/lede/staging_dir/target-aarch64_generic_musl/stamp/.package_compile' failed
make[1]: *** [/home/ling/lede/staging_dir/target-aarch64_generic_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/ling/lede'
/home/ling/lede/include/toplevel.mk:222: recipe for target 'world' failed
make: *** [world] Error 2

--- a/package/libs/mbedtls/Makefile
+++ b/package/libs/mbedtls/Makefile
@@ -122,7 +122,7 @@ PKG_INSTALL:=1
 
 TARGET_CFLAGS += -ffunction-sections -fdata-sections
 TARGET_CFLAGS := $(filter-out -O%,$(TARGET_CFLAGS))
-ifneq ($(CONFIG_LIBMBEDTILS_HAVE_ARMV8CE_AES),)
+ifneq ($(CONFIG_LIBMBEDTLS_HAVE_ARMV8CE_AES),)
   TARGET_CFLAGS := $(filter-out -march=%,$(TARGET_CFLAGS)) -march=armv8-a+crypto
 endif
 

